homeassistant/components/openhome/media_player.py

```diff
@@ -41,6 +44,37 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the configuration config entry."""
+
+    _LOGGER.debug("Setting up config entry: %s", config_entry.unique_id)
+
+    device = await hass.async_add_executor_job(Device, config_entry.data[CONF_HOST])
```

Comment:
This connection should be done in `__init__.py`. When you cannot connect: `raise ConfigEntryNotReady` and Home Assistant will automatically retry the setup later.

Reply:
Thanks - have made the change and now move initialise the device in `__init__` raising `ConfigEntryNotReady`. 

Quite often I leave these devices in a powered off state for days on end - will there be any issue in this case? As they're SSDP, once powered on, they'll announce themselves.

Reply:
Nope, that’s totally fine.